internal/utils/copy/copy_test.go

```diff
@@ -12,31 +12,52 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package copy
+package copy_test
 
 import (
-	"fmt"
 	"os"
 	"path/filepath"
 	"testing"
 
+	"github.com/ZupIT/horusec/internal/utils/copy"
+	"github.com/ZupIT/horusec/internal/utils/testutil"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestCopy(t *testing.T) {
-	t.Run("Should success copy dir", func(t *testing.T) {
-		srcPath, err := filepath.Abs("../../../config")
-		assert.NoError(t, err)
+	src := testutil.GoExample1
+	dst := filepath.Join(t.TempDir(), t.Name())
 
-		dstPath, err := filepath.Abs(".")
-		assert.NoError(t, err)
+	tmpFile, err := os.CreateTemp(os.TempDir(), "test-symlink")
+	require.Nil(t, err, "Expected nil error to create temp file")
 
-		dstPath = fmt.Sprintf(dstPath+"%s", "/tmp-test")
+	symlinkFile := filepath.Join(src, "symlink")
+	err = os.Symlink(tmpFile.Name(), symlinkFile)
+	require.NoError(t, err, "Expected nil error to create symlink file: %v", err)
 
-		err = Copy(srcPath, dstPath, func(src string) bool { return false })
-		assert.NoError(t, err)
+	t.Cleanup(func() {
+		err := os.Remove(symlinkFile)
+		assert.NoError(t, err, "Expected nil error to clean up symlink file: %v", err)
+	})
 
-		err = os.RemoveAll(dstPath)
-		assert.NoError(t, err)
+	err = copy.Copy(src, dst, func(src string) bool {
+		ext := filepath.Ext(src)
+		return ext == ".mod" || ext == ".sum"
 	})
+
+	assert.NoError(t, err)
+
+	assert.DirExists(t, dst)
+	assert.DirExists(t, filepath.Join(dst, "api", "routes"))
+	assert.DirExists(t, filepath.Join(dst, "api", "util"))
+
+	assert.NoFileExists(t, filepath.Join(dst, "symlink"))
+	assert.FileExists(t, filepath.Join(dst, "api", "server.go"))
+	assert.FileExists(t, filepath.Join(dst, "api", "routes", "healthcheck.go"))
+	assert.FileExists(t, filepath.Join(dst, "api", "util", "util.go"))
+
+	assert.NoFileExists(t, filepath.Join(dst, "go.mod"))
+	assert.NoFileExists(t, filepath.Join(dst, "go.sum"))
+
 }
```

Comment:
there is a need for failed cases tests?

Reply:
I don't understand your point here.

Reply:
there is just one test for copy, maybe do some tests that covers failed attempts

Reply:
I don't see any advantage on testing fail scenarios here. Basically all fails will be related with SO, file does not exists, file already exists, etc. Mocking the SO syscalls will introduce a lot of complexity to a simple piece of code.

Reply:
i agree